pybossa/view/bloomberg.py

```diff
@@ -83,11 +83,13 @@ def handle_bloomberg_response():
             # User is authenticated on BSSO, but does not yet have a GIGwork account, auto create one.
             user_data = {}
             try:
-                user_data['fullname']   = attributes['firstName'][0] + " " + attributes['lastName'][0]
-                user_data['email_addr'] = attributes['emailAddress'][0]
-                user_data['name']       = attributes['username'][0]
-                user_data['data_access']= ["L4"]
-                user_data['password']   = generate_password()
+                user_data['fullname']    = attributes['firstName'][0] + " " + attributes['lastName'][0]
+                user_data['email_addr']  = attributes['emailAddress'][0]
+                user_data['name']        = attributes['username'][0]
+                user_data['password']    = generate_password()
+                user_data['metadata']    = {"admin" : "BSSO"}
+                user_data['user_type']   = get_user_type(attributes.get('firmId', ['None'])[0])
```

Comment:
in `def get_user_type(firm_num):` and `get_user_data_access_level(firm_num):`
firm number is converted to int so 'None' or 'L4'(below) may cause issues

Reply:
Ok testing this further

Reply:
I'm adding a check to verify that firm_num is not "None" before the line that converts to int